lupa/_lupa.pyx

```diff
@@ -685,7 +726,8 @@ cdef class _LuaObject:
                     "item/attribute access not supported on functions")
             # table[nil] fails, so map None -> python.none for Lua tables
             py_to_lua(self._runtime, L, name, wrap_none=lua_type == lua.LUA_TTABLE)
-            lua.lua_gettable(L, -2)
+            if not self._gettable(old_top + 1, old_top + 2):
+                raise LuaError("table field access error")
```

Comment:
Can't we wrap `lua_gettable()` in a C function that uses `setjmp()` to catch the error?

Reply:
It's not clear to me how would we be able to catch the error bzw the call to `longjmp` in C.

Reply:
Ok, I read some more on error handling in Lua C-API code. Seems that Lua's `longjmp` error handling is not exposed by itself and things _are_ really complicated. `lua_atpanic()` seems an option here, but requires some form of runtime global C state to figure out where to jump back to. Or use protected mode, as you did. :-[

Reply:
Right, currently we call `lua_atpanic(1)`, altough I am not quite sure what it does when passing 1 (is it stack indexation?).
I believe the only safe way is to call in protected mode, unfortunately. I've read in http://lua-users.org/wiki/ErrorHandlingBetweenLuaAndCplusplus that we could compile Lua so that instead of C `longjmp`, C++ exceptions are raised. But I guess we could only do that with bundled Lua, not with pkg-config Lua or custom Lua.